### PR TITLE
chore(sink): exclude java iceberg and deltalake sink from connector node

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Exclude the module of java `iceberg` and `deltalake` sink from connector node.

The previous `deltalake_rust` will take over the current java version of delta lake and deltalake sink created with `connector = 'deltalake'` will use the rust implementation now.

The package size reduced from 407,850,813 to 119,393,608. Fix https://github.com/risingwavelabs/risingwave/issues/13113.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

1. The previous `connector = 'iceberg_java'` is deprecated, and users can only create sink with rust version of iceberg.
2. The deltalake sink will use the rust version implementation instead. 

</details>

### CODE REVIEW

File: e2e_test/sink/deltalake_rust_sink.slt [hunk 1]
```diff
@@ -7,7 +7,7 @@ CREATE MATERIALIZED VIEW mv6 AS SELECT * FROM t6;
 statement ok
 create sink s6 as select * from mv6
 with (
-    connector = 'deltalake_rust',
+    connector = 'deltalake',
```
Comment (tabVersion) [inline e2e_test/sink/deltalake_rust_sink.slt:10]: Is it a user facing change?

File: src/frontend/src/optimizer/plan_node/stream_sink.rs [hunk 2]
```diff
@@ -150,13 +149,6 @@ impl StreamSink {
             Distribution::Single => RequiredDist::single(),
             _ => {
                 match properties.get("connector") {
-                    Some(s) if s == "deltalake" => {
-                        // iceberg with multiple parallelism will fail easily with concurrent commit
-                        // on metadata
-                        // TODO: reset iceberg sink to have multiple parallelism
```
Comment (tabVersion) [inline src/frontend/src/optimizer/plan_node/stream_sink.rs:156]: What's the parallelism requirement  for iceberg rust version

File: ci/scripts/connector-node-integration-test.sh [hunk 3]
```diff
@@ -115,43 +115,43 @@ for ((i=0; i<${#type[@]}; i++)); do
...
+#    cd ${RISINGWAVE_ROOT}/java/connector-node/python-client
+#    python3 pyspark-util.py create_iceberg
+#    if python3 integration_tests.py --upsert_iceberg_sink ${upsert_sink_input_feature[i]}; then
+#      python3 pyspark-util.py test_upsert_iceberg --input_file="./data/upsert_sink_input.json"
+#      echo "Upsert iceberg sink ${type[i]} test passed"
+#    else
+#      echo "Upsert iceberg sink ${type[i]} test failed"
+#      exit 1
+#    fi
+#    python3 pyspark-util.py drop_iceberg
```
Comment (tabVersion) [inline ci/scripts/connector-node-integration-test.sh:128]: May remove the commands

**Review by tabVersion — COMMENTED** [review]
Nice to switch the connector to rust version.

**Review by tabVersion — APPROVED** [review]

File: src/frontend/src/optimizer/plan_node/stream_sink.rs (hunk 2 above)
Comment (wenym1) [reply src/frontend/src/optimizer/plan_node/stream_sink.rs:156]: The comment was for previous iceberg sink. Now there is no singleton requirement on both iceberg sink and delta lake sink.

**Review by ZENOTME — APPROVED** [review]

**Review by liurenjie1024 — APPROVED** [review]
LGTM. BTW, why we don't remove all codes rather just remove the packaging?

**Review by xxhZs — APPROVED** [review]

**Comment by wenym1** [comment]
> LGTM. BTW, why we don't remove all codes rather just remove the packaging?

Just save for reference given that there is a simple way to exclude it from packaging. We can remove the code when it needs extra effort to maintain in the future.

**Pull request merged**